akka-remote/src/test/scala/akka/remote/artery/TwoNodeArterySpec.scala

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import akka.actor.{ ActorSystem, ExtendedActorSystem }
+import akka.testkit.AkkaSpec
+import com.typesafe.config.{ Config, ConfigFactory }
+
+object TwoNodeArterySpec {
+  def defaultConfig =
+    ConfigFactory.parseString("""
+      akka {
+        actor.provider = "akka.remote.RemoteActorRefProvider"
+        actor.warn-about-java-serializer-usage = off
+        remote.artery {
+          enabled = on
+          hostname = localhost
+          port = 0
+        }
+      }
+    """)
+}
+
+/**
+ * Base class for remoting tests what needs to test interaction between two actor systems
+ * communicating over artery.
+ */
+abstract class TwoNodeArterySpec(config: Config) extends AkkaSpec(config) {
```

Comment:
Yeah, things are messy in these tests. In the end these abstractions hurt more than help as they become spaghetti of special cases overriding stuff.